jdat_notebooks/preimaging/preimaging_03_association.ipynb

```diff
@@ -0,0 +1,454 @@
+    "        (80.4792, -69.5564)], color='yellow')\n",
+    "    \n",
+    "    ax.add_patch(rect1)\n",
+    "    \n",
+    "    rect2 = leopolygon(current_detector_coords, edgecolor='navy', \n",
+    "                       facecolor='navy', alpha=0.8)\n",
+    "    \n",
+    "    new_shape = so.cascaded_union(referencia)\n",
+    "    xs, ys = new_shape.exterior.xy\n",
+    "    \n",
+    "    ax.fill(xs, ys, alpha=0.8, fc='red', ec='red')\n",
+    "\n",
+    "    ax.add_patch(rect2)\n",
+    "    \n",
+    "    plt.xlim([80.2, 80.8]) # RA\n",
+    "    plt.ylim([-69.4, -69.6]) # Dec\n",
+    "    \n",
+    "    plt.title('coverage = {}% \\n {}'.format(str(covfile[current_detector_idx]), \n",
+    "                                            current_detector), fontsize=15)\n",
+    "    plt.xlabel('RA (degrees)')\n",
+    "    plt.ylabel('Dec (degrees)')\n",
+    "    \n",
+    "    pdfname = 'figure_' + str(len(finallist)-1)\n",
+    "    # plt.savefig(pdfname) \n",
+    "  \n",
+    "    plt.show()\n",
+    "     \n",
+    "\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Here we need to go through the final list and calculate the overlap \n",
+    "with the stellar catalogue. We select the files that have an overlap of 99% or larger.\n",
+    "\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "bestlist = []\n",
+    "# these coordinates represent then extend of the LMC catalogue\n",
+    "referencia = Polygon([(80.3125, -69.4950), \n",
+    "                      (80.4958, -69.4392), \n",
+    "                      (80.6625, -69.5022), \n",
+    "                      (80.4792, -69.5564)])\n",
+    "\n",
+    "rejected = 0\n",
+    "for current_detector, current_detector_coords in zip(finallist, finalcoords):\n",
+    "    detector = Polygon(current_detector_coords)\n",
+    "    \n",
+    "    coverage = detector.intersection(referencia).area\n",
+    "    coverage = round(100 * coverage / detector.area)\n",
+    "    \n",
+    "    if coverage >= 99:\n",
+    "        bestlist.append(current_detector)\n",
+    "    else:\n",
+    "        rejected += 1\n",
+    "        \n",
+    "print('number of overlapping images = ', len(bestlist))\n",
+    "print('number of rejected images = ', rejected)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Save Detector List to an Association (JSON file)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# here we print a json file with the science files in the correct order\n",
+    "\n",
+    "association = {}\n",
+    " \n",
+    "association[\"asn_id\"] = \"a3001\"\n",
+    "association[\"asn_pool\"] = \"none\" \n",
+    "association[\"asn_rule\"] = \"Asn_Image\" \n",
+    "association[\"program\"] = \"1069\" \n",
```

Comment:
There is a way to programmatically create association files using functions in the `jwst` package. I haven't tried it out, but it was added I believe in version 0.16.0.